dexcom/client/client.go

```diff
@@ -26,9 +29,6 @@ func New(cfg *client.Config, tknSrcSrc oauth.TokenSourceSource) (*Client, error)
 		return nil, err
 	}
 
-	// NOTE: Dexcom authorization server does not support HTTP Basic authentication
```

Comment:
This is a no-op, found via staticcheck.

Reply:
When this was first implemented this was required because the Dexcom API did not correctly conform to the OAuth2 spec and this allowed the OAuth library to work with Dexcom. (I don't remember exactly what was broken, though.) Since we've updated this dependency since then and the function has become a no-op, and the Dexcom API appears to still work, it appears we can remove it. (Though, I suppose, in theory, this could be the cause of some of the Dexcom failures.)